test/extended/builds/run_policy.go

```diff
@@ -0,0 +1,241 @@
+package builds
+
+import (
+	"fmt"
+	"strings"
+
+	g "github.com/onsi/ginkgo"
+	o "github.com/onsi/gomega"
+
+	buildapi "github.com/openshift/origin/pkg/build/api"
+	buildclient "github.com/openshift/origin/pkg/build/client"
+	buildutil "github.com/openshift/origin/pkg/build/util"
+	exutil "github.com/openshift/origin/test/extended/util"
+	kapi "k8s.io/kubernetes/pkg/api"
+)
+
+var _ = g.Describe("[builds][Slow] using build configuration runPolicy", func() {
```

Comment:
@smarterclayton each of this tests take ~20s to complete, but sure, I will add some 'fast' running build test. Also I have couple unit tests to verify the different run policy strategies on unit level to make sure we pass the build in correct order.